elftools/elf/dynamic.py

```diff
@@ -189,6 +250,39 @@ def __init__(self, header, stream, elffile):
                 break
         Segment.__init__(self, header, stream)
         Dynamic.__init__(self, stream, elffile, stringtable, self['p_offset'])
+        self._symbol_index_map = None
+        self._symbol_name_map = None
+        self._num_symbols = None
+
+    def num_symbols(self):
+        """ Number of symbols in the table recovered from DT_SYMTAB
+        """
+        if self._num_symbols is None:
+            self._num_symbols = len(list(self.iter_symbols()))
+        return self._num_symbols
+
+    def get_symbol(self, index):
+        """ Get the symbol at index #index from the table (Symbol object)
+        """
+        if self._symbol_index_map is None:
+            self._symbol_index_map = {}
+            for idx, symbol in enumerate(self.iter_symbols()):
+                self._symbol_index_map[idx] = symbol
+        return self._symbol_index_map[index]
```

Comment:
If symbol_index_map is strictly a map from index to symbol, it is probably better as a list - then this whole thing can be a list comprehension

Reply:
Oh, indeed, I had written it differently earlier and changed it without properly rethinking... I'll do that, thanks!